statsforecast/core.py

```diff
@@ -1546,6 +1551,113 @@ def plot(
             palette="tab20b",
         )
 
+    def save(
+        self,
+        path: Union[Path, str],
+        file_name: Union[Path, str, None] = None,
+        max_size: str = "50MB",
+        trim: bool = True,
+    ):
+        """Function that will save StatsForecast class with certain settings to make it
+        reproducible.
+
+        Parameters
+        ----------
+        path: Union[Path, str]
+            Path of the folder where the pickle should be saved to
+        file_name: Union[Path, str, None] = None
+            Name of the file, if None then datetime name will be given
+        max_size: str = "50MB"
+            StatsForecast class should not exceed this size, available byte naming:
+            ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
+        trim: bool = True
+            Delete any attributes not needed for inference
+        """
+
+        # Removing unnecessary attributes
+
+        # @jmoralez decide future implementation
+        trim_attr: list = ["fcst_fitted_values_", "cv_fitted_values_"]
+        if trim:
+            for attr in trim_attr:
+                # remove unnecessary attributes here
+                self.__dict__.pop(attr, None)
+
+        # Will be used to find the size of the fitted models
+        # Never expecting anything higher than GB (even that's a lot')
+        bytes_hmap = {
+            "B": 1,
+            "KB": 2**10,
+            "MB": 2**20,
+            "GB": 2**30,
+            "TB": 2**40,
+            "PB": 2**50,
+            "EB": 2**60,
+            "ZB": 2**70,
+            "YB": 2**80,
+        }
+
+        max_size = max_size.replace(" ", "")
+        match = re.match(r"(\d+\.\d+|\d+)(\w+)", max_size)
+        if not match or match[2] not in bytes_hmap.keys():
+            warnings.warn("Couldn't parse your max_size, default 50MB will be set")
+            cap_size = 50.0 * bytes_hmap["MB"]
+        else:
+            m_size = float(match[1])
+            key_ = match[2]
+            cap_size = m_size * bytes_hmap[key_]
+
+        sf_size = len(pickle.dumps(self))
+
+        if sf_size >= cap_size:
+            return
+
+        converted_size, sf_byte = None, None
+        for key in reversed(bytes_hmap.keys()):
+            x_byte = bytes_hmap[key]
+            if sf_size >= x_byte:
+                converted_size = sf_size / x_byte
+                sf_byte = key
+                break
+
+        if converted_size is None or sf_byte is None:
+            error_msg = "Internal Error, this shouldn't happen, please open an issue"
+            raise NameError(error_msg)
+
+        print(f"Model(s) size: {converted_size:.2f}{sf_byte}")
+
+        print("Saving model(s)")
+
+        if not file_name:
+            datetime_record = dt.datetime.now().strftime("%m_%d_%Y_%H_%M_%S")
```

Comment:
We should use a more standard format here. Also using UTC time removes ambiguities about the timezone of the machine running this code.
```suggestion
            datetime_record = dt.datetime.utcnow().strftime("%Y-%m-%d_%H:%M:%S")
```

Reply:
this format isn't allowed: https://www.mtu.edu/umc/services/websites/writing/characters-avoid/#:~:text=Illegal%20Filename%20Characters&text=Don%27t%20start%20or%20end,underscores%3B%20use%20a%20hyphen%20instead
we can do:
`"%Y_%m_%d_%H_%M_%S"` or `"%Y-%m-%d-%H-%M-%S"` or even `"%Y-%m-%d_%H-%M-%S"`

Reply:
Let's go with "%Y-%m-%d_%H-%M-%S"